src/time_dependent_operator.jl

```diff
@@ -0,0 +1,302 @@
+
+import Base: size, *, +, -, /, ==, isequal, adjoint, convert
+
+abstract type AbstractTimeDependentOperator{BL,BR} <: AbstractOperator{BL,BR} end
+
+current_time(::AbstractOperator) = throw(ArgumentError("Time not defined for operator."))
+static_operator(o::AbstractOperator) = o
+
+set_time!(o::AbstractOperator, ::Number) = o
+set_time!(o::LazyOperator, t::Number) = (set_time!.(o.operators, t); return o)
+
+(o::AbstractTimeDependentOperator)(t::Number) = set_time!(o, t)
+
+function _check_same_time(A::AbstractTimeDependentOperator, B::AbstractTimeDependentOperator)
+    current_time(A) == current_time(B) || throw(ArgumentError("Time-dependent operators with different times cannot be combined."))
+end
+
+for func in (:basis, :length, :size, :tr, :normalize, :normalize!,
+    :identityoperator, :one, :eltype, :ptrace)
+    @eval $func(op::AbstractTimeDependentOperator) = $func(static_operator(op))
+end
+
+for func in (:expect, :variance)
+    @eval $func(op::AbstractTimeDependentOperator{B,B}, x::Ket{B}) where B = $func(static_operator(op), x)
+    @eval $func(op::AbstractTimeDependentOperator{B,B}, x::AbstractOperator{B,B}) where B = $func(static_operator(op), x)
+    @eval $func(index::Integer, op::AbstractTimeDependentOperator{B1,B2}, x::AbstractOperator{B3,B3}) where {B1,B2,B3<:CompositeBasis} = $func(index, static_operator(op), x)
+    @eval $func(indices, op::AbstractTimeDependentOperator{B1,B2}, x::AbstractOperator{B3,B3}) where {B1,B2,B3<:CompositeBasis} = $func(indices, static_operator(op), x)
+end
+
+promote_rule(::Type{T}, ::Type{S}) where {T<:AbstractTimeDependentOperator,S<:AbstractOperator} = T
+convert(::Type{T}, O::AbstractOperator) where {T<:AbstractTimeDependentOperator} = T(O)
+
+const VecOrTuple = Union{Tuple,AbstractVector}
+
+"""
+    TimeDependentSum(lazysum, coeffs, init_time)
+    TimeDependentSum(::Type{Tf}, basis_l, basis_r; init_time=0.0)
+    TimeDependentSum([::Type{Tf},] [basis_l,] [basis_r,] coeffs, operators; init_time=0.0)
+    TimeDependentSum([::Type{Tf},] coeff1=>op1, coeff2=>op2, ...; init_time=0.0)
+    TimeDependentSum(::Tuple, op::TimeDependentSum)
+
+Lazy sum of operators with time-dependent coefficients. Wraps a `LazySum` `lazysum`,
```

Comment:
May we do `@ref` links for these references so that they are linked in the documentation?